Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -185,5 +183,5 @@ public struct SwiftlyHTTPClient {
 }
 
 private class HTTPClientWrapper {
-    fileprivate let inner = HTTPClient.shared
+    fileprivate let inner = HTTPClient(eventLoopGroupProvider: .singleton)
```

Comment:
Note that this was added because in ad hoc testing I discovered an emergent http decompression limit error. It appears that something has changed to trigger it in the GitHub API's.

Reply:
Does `.shared` have different limits to the HTTPClient with a default configuration?

Reply:
It appears so. I dug a little deeper because I was concerned that this could be affecting everyone using swiftly. This is the commit that regressed and the testing didn't pick it up at the time: 2d054143ea29e2e5ee0085cbb0ca2ed8086d9166

I'm fully reverting that change so that this is not broken, but it does leave some open questions about our commit verification process. Ideally, some form integration testing might have picked this up at the time. Also, it does seem suspect that the shared HTTP Client has a broken decompression limit, whereas this initializer that just reuses the singleton event loop group provider has an intact decompression.

Reply:
Have you reported this in the async-http-client repository?

Reply:
Including @weissi as I believe this was from his PR to use shared

Reply:
I can try and create a small reproducible test case and see if the latest fixes this. I hope to do that once I've restored this to a working state here with the original working code.

Reply:
@adam-fowler the latest commit fully reverts the HTTPClient.swift portion of 2d0514 and smoke testing seems to be working again. Any chance that we can get these fixes in to get us back into a working state?

Reply:
We've run into this problem over in Penny (Vapor's Discord bot) - this was the fix Mahdi implemented: https://github.com/vapor/penny-bot/blob/main/Sources/Shared/%2BHTTPClient.swift#L12-L14

Reply:
Yes, and there is already an issue for it (which I should have marked as completed, but forgot):
https://github.com/swift-server/async-http-client/issues/739

@cmcgee1024 feel free to comment there if you think the limit is still too low for Swiftly, although I'm not sure how much success you'll have with getting the team to agree on increasing it, considering increasing the limit has some implications.

Reply:
Thanks @gwynne 

I think that unless there's a reason why we can't revert the HTTPClient change here, we should probably go with that for now until this issue is fixed in async-http-client: https://github.com/swift-server/async-http-client/issues/739

Reply:
Having investigated, the reason `HTTPClient(eventLoopGroupProvider: .singleton)` works and `HTTPClient.shared` doesn't is the first disables decompression. I would rather we still ran with decompression enabled. If the current rate is too low then let's do what @MahdiBM suggests. I would also like this implemented as a singleton so we don't have to worry about shutdown.

Reply:
@adam-fowler heads up: you've approved this PR, but left a change request in this comment afterwards.

Reply:
>Ideally, some form integration testing might have picked this up at the time.

It appears maybe this was some change on the GitHub side that introduced this issue. I've just tried to run a basic install locally and hit the error, so I'd imagine if we had CI set up right now we'd see it. But yeah, it does indicate that we should be running tests on some regular interval (daily?) to ensure things like this can be caught as they happen.

Reply:
I agree with @adam-fowler here let's define a static one so we don't need the wrapper with the `deinit`

Reply:
> It appears maybe this was some change on the GitHub side that introduced this issue. I've just tried to run a basic install locally and hit the error, so I'd imagine if we had CI set up right now we'd see it. But yeah, it does indicate that we should be running tests on some regular interval (daily?) to ensure things like this can be caught as they happen.

After thinking on this a bit, I don't think that GitHub is the variable as I had initially thought. Everything was working for me until I rebased this change on top of the above commit that changed the way that HTTPClient was initialized. A CI with at least a smoke test that did a `swiftly install latest` would have picked up the problem.

Reply:
>  I would rather we still ran with decompression enabled. If the current rate is too low then let's create a client with a higher rate. I would also like this implemented as a singleton so we don't have to worry about shutdown. eg

Is this blocking the approval and merge (I don't merge permissions here) of this PR? Can we defer this to a new issue/PR? It appears that the existing approach before the regression was working for quite some time and is currently working for everyone on v0.3.0. I'm just restoring that so that we're all unblocked.

Reply:
> Is this blocking the approval and merge (I don't merge permissions here) of this PR? Can we defer this to a new issue/PR?

I would rather we stick to a singleton. The change I suggested works, as I used it when testing the macOS support PR

Reply:
When you say 'a singleton', you mean `HTTPClient.shared` or your own singleton?

I very much think it should be `HTTPClient.shared`, there's no reason to create singletons all over the place, just wastes resources. The design goal is that `HTTPClient.shared` behaves like the platform browser: If Safari/Chrome/... can open it in defaults settings then `HTTPClient.shared` should do.

Reply:
@weissi I understand that but we cannot use `HTTPClient.shared` at the moment. It just keeps throwing errors about compression limits.

Reply:
@adam-fowler I've updated this PR with your suggested change to the construction of the http client.

Reply:
@adam-fowler Correct right this instant but the fix for this is merged, I'm working with the maintainers on cutting a release

Reply:
Thanks to @lukasa, we've got https://github.com/swift-server/async-http-client/releases/tag/1.21.2 now. So `from: "1.21.2"` for AHC should fix this.

Reply:
@weissi / @Lukasa thanks for pushing that through!

>After thinking on this a bit, I don't think that GitHub is the variable as I had initially thought. Everything was working for me until I rebased this change on top of the above commit that changed the way that HTTPClient was initialized. A CI with at least a smoke test that did a swiftly install latest would have picked up the problem.

Prior to the move to swiftlang, we did run tests in CI, a few of which covered that case (`testInstallLatest` and `default-install.sh` in particular). That's what leads me to believe maybe something changed on the GitHub side. @shahmishal is working on getting CI set up under the new organization, so we'll have automated testing back shortly.